components/scream/src/physics/p3/p3_functions_autoconversion_impl.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef P3_FUNCTIONS_AUTOCONVERSION_IMPL_HPP
+#define P3_FUNCTIONS_AUTOCONVERSION_IMPL_HPP
+
+#include "p3_functions.hpp"
+#include "p3_constants.hpp"
+
+namespace scream {
+namespace p3 {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::cloud_water_autoconversion(const Spack& rho, const Spack& inv_rho, const Spack& qc_incld, const Spack& nc_incld,
+    const Spack& qr_incld, const Spack& mu_c, const Spack& nu, Spack& qcaut, Spack& ncautc, Spack& ncautr)
+{
+    // Khroutdinov and Kogan (2000)
+    const auto qc_not_small = qc_incld >=  1e-8;
+    if(qc_not_small.any()){
+        qcaut.set(qc_not_small,
+            sp(1350.0)*pow(qc_incld,sp(2.47))*pow(nc_incld*sp(1.e-6)*rho,sp(-1.79)));
+        // note: ncautr is change in Nr; ncautc is change in Nc
+        ncautr.set(qc_not_small, qcaut*C::CONS3);
+        ncautc.set(qc_not_small, qcaut*nc_incld/qc_incld);
+    }
+
+    const auto qcaut_is_zero = qcaut == 0;
+    if(qcaut_is_zero.any()){ncautc.set(qcaut_is_zero,  0.0);}
```

Comment:
For one line computations, don't call `any()` first; just do the computation with the masked set. In this case, you're just setting masked values to 0, so there's even more reason to avoid the unnecessary conditional.

Reply:
Good point. Thanks